packages/ember-data/lib/system/snapshot.js

```diff
@@ -392,6 +392,16 @@ Snapshot.prototype = {
   },
 
   /**
+    @method serialize
+    @param {Object} options
+    @return {Object} an object whose values are primitive JSON values only
+   */
+  serialize: function(options) {
+    return this.record.store.serializerFor(this.modelName).
```

Comment:
I thought snapshot.serialize should serialize data from snapshot, not from underlying record. See my test below it won't pass if I do as you wrote.